src/test/java/org/ahocorasick/trie/StateTest.java

```diff
@@ -19,6 +22,52 @@ public void constructSequenceOfCharacters() {
         assertEquals(2, currentState.getDepth());
         currentState = currentState.nextState('c');
         assertEquals(3, currentState.getDepth());
+        currentState = currentState.nextState('F');
+        assertNull(currentState);
+    }
+
+    @Test
+    public void getStates() {
+        State rootState = new State();
+        rootState.addState("foo");
+
+        State currentState = rootState.nextState('f');
+
+        Collection<State> states = rootState.getStates();
+        assertEquals(1, states.size());
+        assertEquals(currentState, states.iterator().next());
+    }
+
+
+    @Test
+    public void getTransitions() {
+        State rootState = new State();
+        rootState.addState("foo");
+        State currentState = rootState.nextState('f');
+
+        Collection<Character> transitions = rootState.getTransitions();
+        assertEquals(1, transitions.size());
+        assertEquals(Character.valueOf('f'), transitions.iterator().next());
+    }
+
+    @Test
+    public void failure() {
+        State failureState = new State();
+        State rootState = new State();
+        rootState.setFailure(failureState);
+
+        assertEquals(failureState, rootState.failure());
+
```

Comment:
Style nit: No newline before the closing brace inside methods.

Reply:
👍